logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ClientRepository.kt

```diff
@@ -1,11 +1,49 @@
 package com.wire.kalium.logic.data.client
 
 import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.remote.ClientRemoteRepository
 import com.wire.kalium.logic.data.conversation.ClientId
 import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.persistence.client.ClientRegistrationStorage
 
 interface ClientRepository {
     suspend fun registerClient(param: RegisterClientParam): Either<CoreFailure, Client>
     suspend fun persistClientId(clientId: ClientId): Either<CoreFailure, Unit>
     suspend fun currentClientId(): Either<CoreFailure, ClientId>
+    suspend fun deleteClient(param: DeleteClientParam): Either<CoreFailure, Unit>
+    suspend fun selfListOfClients(): Either<CoreFailure, List<Client>>
+    suspend fun clientInfo(clientId: ClientId /* = com.wire.kalium.logic.data.id.PlainId */): Either<CoreFailure, Client>
+}
+
+
+class ClientDataSource(
+    private val clientRemoteRepository: ClientRemoteRepository,
+    private val clientRegistrationStorage: ClientRegistrationStorage
+) : ClientRepository {
+    override suspend fun registerClient(param: RegisterClientParam): Either<CoreFailure, Client> {
+        return clientRemoteRepository.registerClient(param)
+    }
+
+    override suspend fun persistClientId(clientId: ClientId): Either<CoreFailure, Unit> {
+        clientRegistrationStorage.registeredClientId = clientId.value
```

Comment:
This makes me think... we are never catching DB Exceptions.

Maybe in the future we could wrap repositories' operations that touch the DB to catch them and return do something like:

```kotlin
fun wrapDBOperation(action: () -> T): Either<CoreFailure, T> {
    return try {
        Either.Right(action())
    } catch(dbException: SQLException){
        Either.Left(CoreFailure.MrUserIDontFeelSoGood)
    }
}
```